lymph/midline.py

```diff
@@ -375,9 +597,79 @@ def check_and_assign(self, new_params: np.ndarray):
             raise ValueError(
                 "Spread probs must be between 0 and 1"
             )
+        if 0. > self.midext_prob or self.midext_prob > 1.:
+            raise ValueError(
+                "Midline extension probability must be between 0 and 1"
+            )
 
         self.spread_probs = new_spread_probs
 
+    def _likelihood_mid(
+        self,
+        log: bool = True
+    ) -> float:
+        """Compute the (log-)likelihood of data, using the stored spread probs and
+        fixed distributions for marginalizing over diagnose times.
+
+        This method mainly exists so that the checking and assigning of the
+        spread probs can be skipped.
+        """
+        stored_t_stages = set(self.ipsi.diagnose_matrices.keys())
+        provided_t_stages = set(self.ipsi.diag_time_dists.keys())
+        t_stages = list(stored_t_stages.intersection(provided_t_stages))
+        
+        max_t = self.diag_time_dists.max_t
+        state_probs_ex = np.zeros(shape=len(self.state_list), dtype=float)
+        state_probs_ex[0] = 1.
+        state_probs_nox = np.zeros(shape=len(self.state_list), dtype=float)
+        state_probs_nox[0] = 1.
+        state_probs_midext = np.zeros(shape=len(self.midexstate_list), dtype=float)
+        state_probs_midext[0] = 1.
```

Comment:
_ex is for the state vector given midline extension and _nox for the state vector given no midline extension. _midext is for the midline state vector that is tracked separately.

Oh, yes that is true! Would it make sense to separately initialize the ipsi and contra state probability vectors because they are tracked separately afterwards?

Reply:
Yes, I think so. But not sure 😅

Reply:
I'll try it. Makes the most sense.